zerver/worker/queue_processors.py

```diff
@@ -620,6 +620,11 @@ def consume(self, event: Mapping[str, Any]) -> None:
                                               delete_after_upload=True)
             assert public_url is not None
 
+            # TODO: This enables support for delete after access, and needs to be tested.
+            export_event = RealmAuditLog.objects.get(id=event['id'])
+            export_event.extra_data = public_url
```

Comment:
Is the idea here that the "delete" feature will reconstruct the actual path from `public_url`?  I guess that's arguably more future-safe for cases where the upload backend gets migrated, so maybe that's the correct approach.  But let's actually write that function in the upload_backend classes and make sure it works (ideally, we'd extend our main end-to-end tests to actually do the fake "upload" of the file, verify it can be downloaded using `self.client_get`, delete it using the function for deleting it, and then verify it isn't still served).  That would give test coverage on the full end-to-end experience, which is definitely missing in this PR (e.g. the `public_url` you returned didn't work if you ran the command locally and then pasted the URL into a browser).